pulsar-common/src/main/java/org/apache/pulsar/common/functions/FunctionConfig.java

```diff
@@ -91,4 +91,8 @@
     private Long timeoutMs;
     private String jar;
     private String py;
+
+    // Any non-default artifactory that need to be used to download dependencies
+    // This is going to be Runtime specific.
+    private String artifactory;
```

Comment:
`artifactory` is the name of one specific product to manage the repository. 

In practice, this can be whatever repository. Also should we clarify that this only applies to Python and Java still needs to be packaged with fat-jars?

Renaming into `pythonDepenedenciesRepository` or similar name?

Reply:
I will call it dependencyRepository to make it general enough for later runtimes as well.
Also I'm thinking instead of making it part of FunctionConfig, it should ideally be part of workerConfig. Since only admins can decide where users can download stuff. What do you think?

Reply:
Yes, I missed that detail. WorkerConfig sounds a better place to me. No need for developer to know about it.